spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural_common.py

```diff
@@ -285,17 +286,17 @@ def __write_common_rewiring_data(
         # of the cores that have perform structural updates.
         # NOTE: it should be different between application vertices
         if app_vertex not in self.__seeds.keys():
-            self.__seeds[app_vertex] = \
-                [self.__rng.randint(0x7FFFFFFF) for _ in range(4)]
+            self.__seeds[app_vertex] = (
+                create_mars_kiss_seeds(self.__rng, self.__seed))
 
         # write the random seed (4 words), generated randomly,
         # but the same for all postsynaptic vertices!
         for seed in self.__seeds[app_vertex]:
             spec.write_value(data=seed)
 
         # write local seed (4 words), generated randomly!
-        for _ in range(4):
-            spec.write_value(data=numpy.random.randint(0x7FFFFFFF))
+        for seed in create_mars_kiss_seeds(self.__rng, self.__seed):
+            spec.write_value(data=seed)
```

Comment:
It should be possible to `spec.write_array(create_mars_kiss_seeds(self.__rng, self.__seed))`

Reply:
this is true. will fix

Reply:
fixed